### PR TITLE
 #162633459 Implement Reset Password Functionality

### PR DESCRIPTION
**Pull request by NabunyaLilian**
**What does this PR do?**

Implements reset password functionality.

**Description of Task to be completed?**
- Create components for sending email and reseting password
- Create actions 
- Create reducers
- Test the actions, reducers and components.

**How should this be manually tested?**
- Run `npm install` to install any dependencies
- Run `npm test` to check for testing.
- Run `npm start` to run the application
- On the UI click `sign in` and if you forgot your password click `reset password`

**What are the relevant pivotal tracker stories?**
[#162633459](https://www.pivotaltracker.com/story/show/162633459)

### CODE REVIEW

**Comment by araaliFarooq** [comment]
Good UI, though validation is still lacking.

**Comment by fatukunda** [comment]
@NabunyaLilian The UI looks good but the functionality is still somehow lacking. It shows that a user is logged in and yet they are able to access the reset password page which in my understanding should be accessed if a user forgot their password and therefore not logged in.

**Comment by NabunyaLilian** [comment]
Hey guys thanks for the  feedback, I have worked on it.

File: src/actions/ResetPasswordLinkAction.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+export const passwordResetFail = error => (
+    {
+        type: actionTypes.PASSWORD_RESET_FAILED,
+        payload: error
+    }
+);
+
+export const resetPasswordAction = email => (dispatch) => {
+    const url = 'https://ah-backend-valkyrie-staging.herokuapp.com/api/v1/users/reset_password'
```
Comment (ruganda) [inline src/actions/ResetPasswordLinkAction.js]: Please, Put this in an environment variable

File: src/actions/ChangePasswordAction.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+export const passwordChangeFail = error => (
+  {
+      type: actionTypes.PASSWORD_RESET_FAILED,
+      payload: error
+  }
+);
+
+export const changePasswordAction = (password, token) => (dispatch) => {
+    const url = `https://ah-backend-valkyrie-staging.herokuapp.com/api/v1/users/reset_password_confirm/${token}`
```
Comment (ruganda) [inline src/actions/ChangePasswordAction.js]: Please make use of the environment variables

File: src/components/auth/reset_password.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    this.resetTextInput();
+  }
+
+  resetTextInput(){
+    this.setState({email: ''})
+  }
+
+ render() {
+  const success = this.props.errors.message;
+  console.log(success)
```
Comment (ruganda) [inline src/components/auth/reset_password.js]: Please remove this console log

**Comment by NabunyaLilian** [comment]
Hey @ruganda , thanks for the feedback I'm done working on it.

**Comment by mubstimor** [comment]
LGTM

**Comment by araaliFarooq** [comment]
This LGTM.

**Comment by mycok** [comment]
LGTM

**Pull request merged**